exercises/rail-fence-cipher/examples/success-standard/src/RailFenceCipher.hs

```diff
@@ -0,0 +1,10 @@
+module RailFenceCipher (encode, decode) where
+
+import Data.List (sort)
+
+encode :: (Ord a) => Int -> [a] -> [a]
```

Comment:
Will you explain to me why `(Ord a) =>` necessary? I am not seeing why, but I may have just missed it.

Reply:
Seems like it isn't, I think it was from a previous version

Reply:
Wow my bad there, seems like in the sort is needed, cause it takes on count the second argument.

Reply:
Should I leave it as it is? Or use sortBy?

Reply:
to sum up, I think that there are 2 options:
```Haskell
decode :: Int -> [a] -> [a]
decode n xs = map snd $ sortBy (compare `on` fst) zippedL
  where zippedL = zip (encode n [0..length xs - 1]) xs
```
or what we already have.

Reply:
I see, because you are using `sort` in `decode`.

In contrast to the stub, where I argued that we only need show the student what was specified by the problem description, here in the example I would in fact prefer not to add constraint that isn't necessary. Because the example is doing well by expanding from `String` to some other type. But if it's stopping at `(Ord a)` then it's only going halfway! It should go all the way!

Reply:
Thus, let us use `sortBy` here!